tests/ha/ctdb.pm

```diff
@@ -0,0 +1,121 @@
+# SUSE's openQA tests
+#
+# Copyright (c) 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test ctdb resource agent
+# Maintainer: Julien Adamek <jadamek@suse.com>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use lockapi;
+use hacluster;
+use utils qw(systemctl exec_and_insert_password);
+
+sub run {
+    my $cluster_name = get_cluster_name;
+    my $vip_ip       = '10.0.2.20';
+    my $ctdb_folder  = '/srv/fs_cluster_md/ctdb';
+
+    # CTDB configuration must be done only in cluster nodes
+    if (check_var('CTDB_TEST_ROLE', 'server')) {
+        my $node_01_host = choose_node(1);
+        my $node_02_host = choose_node(2);
+        my $node_01_ip   = get_ip($node_01_host);
+        my $node_02_ip   = get_ip($node_02_host);
+        my $ctdb_cfg     = '/etc/samba/smb.conf';
+        my $ctdb_rsc     = 'ctdb';
+        my $nmb_rsc      = 'nmb';
+        my $smb_rsc      = 'smb';
+        my $ip_rsc       = 'vip';
+
+        # IP addresses of all nodes must be in nodes file
+        if (get_node_number == 3) {
+            my $node_03_host = choose_node(3);
+            my $node_03_ip   = get_ip($node_03_host);
+            assert_script_run "echo -e '$node_01_ip\n$node_02_ip\n$node_03_ip' > /etc/ctdb/nodes";
+        }
+        else {
+            assert_script_run "echo -e '$node_01_ip\n$node_02_ip' > /etc/ctdb/nodes";
+        }
```

Comment:
I think you can simplify this with something like that:

    my @node_list;
    foreach my $node (1 .. get_node_number) {
        my $node_ip = get_ip(choose_node($node));
        push @node_list, $node_ip;
    }
    assert_script_run 'echo -e "' . join("\n", @node_list) . '" > /etc/ctdb/nodes';

(I'm not sure for the last line, an intermediate variable may be needed for the `join("\n", @node_list)` part).

You can them remove the `node_*_ip` and `node_*_host` variables (these variables seem to be used only once).

Reply:
Yes you are right, It's ok now that I adapted get_node_number function [here](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/9241)